jazz_codeq/index.js

```diff
@@ -0,0 +1,415 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =========================================================================
+
+const errorHandlerModule = require("./components/error-handler.js"); //Import the error codes module.
+const responseObj = require("./components/response.js"); //Import the response module.
+const configObj = require("./components/config.js"); //Import the environment data.
+const logger = require("./components/logger.js"); //Import the logging module.
+const secretHandlerModule = require("./components/secret-handler.js"); //Import the secret-handler module.
+const request = require('request');
+const moment = require('moment');
+const async = require('async');
+const messages = require("./config/messages");
+const contextObj = require("./components/context.js"); //Import the context data.
+
+module.exports.handler = (event, context, cb) => {
+
+	//Initializations
+	let serviceContext = contextObj(event);
+	let errorHandler = errorHandlerModule();
+	let config = configObj(serviceContext);
+	let secretHandler = secretHandlerModule();
+	logger.init(serviceContext, context);
+
+	const DATE_FORMAT = 'YYYY-MM-DDThh:mm:ss';
+	const APPEND_ZEROES = '-0000';
+
+	try {
+
+		let pathString;
+		logger.info(serviceContext.resourcePath);
+		
+		getAPIPath(serviceContext.resourcePath)
+		.then(result => {
+			pathString = data.pathString;
+		}).catch(err => {
+			cb(JSON.stringify(errorHandler.throwInputValidationError(err.errorMessage)));
+		});
+
+		logger.info(pathString);
+
+		if (serviceContext && serviceContext.method && serviceContext.method === 'GET' && pathString === "codeq") {
+			logger.info("code quality service called.");
+
+			let required_fields = config.required_params.map(v => v.toLowerCase());
+			let query = getQuery(serviceContext);
+
+			let fromDate;
+			let toDate;
+
+			// validate required fields
+			let missing_required_fields = required_fields.filter(key => !query[key]);
+
+			if (missing_required_fields.length) {
+				message = messages.MISSING_FIELDS + missing_required_fields.join(", ");
+				logger.error(message);
+				return cb(JSON.stringify(errorHandler.throwInputValidationError(message)));
+			}
+
+			let metrics = getMetrics(query);
+
+			//if from and to dates are provided validate, else assign default dates
+			if (query.from) {
+				if (!moment(query.from, moment.ISO_8601, true).isValid()) {
+					return cb(JSON.stringify(errorHandler.throwInputValidationError(messages.INVALID_FROM_DATE)));
+				} else {
+					fromDate = moment(query.from).format(DATE_FORMAT) + APPEND_ZEROES;
+				}
+			} else {
+				fromDate = moment().subtract(1, "days").format(DATE_FORMAT) + APPEND_ZEROES;
+			}
+
+			if (query.to) {

```

Comment:
It will be better if we put these validations inside a method so that it also unit testable